pandas/core/frame.py

```diff
@@ -7387,7 +7387,10 @@ def _to_arrays(data, columns, coerce_float=False, dtype=None):
         if isinstance(data, np.ndarray):
             columns = data.dtype.names
             if columns is not None:
-                return [[]] * len(columns), columns
+                arrays = [np.array([], dtype=dtype)
+                          for _, dtype in data.dtype.descr]
```

Comment:
What's the reasoning for using the `descr` here instead of just the dtype? ref #20734 and the docs for `dtype.descr`I think we may run into some obscure issues using the str representation of the type instead of the type directly